### ISSUE TITLE
Enable NCO and CDO build support

### ISSUE THREAD

**Issue by aashish24** [UV-CDAT/uvcdat#1497]

**Comment by durack1** [comment]
@aashish24 is there any particular reason why this is required? These packages duplicate the functionality already present in UV-CDAT..

**Comment by doutriaux1** [comment]
@durack1 they will be added into cdat and python wrapped. They're very fast and used broadly. @czender is looking into this.

**Comment by aashish24** [comment]
@durack1 yes some of the functionality is duplicated but not all as I understand, plus they are used by a much larger community. @czender any idea when it will get done?

**Comment by czender** [comment]
The ACME WF schedule has this to be completed by 12/31. Python wrappers for NCO work. It's all about build infrastructure and repositories now. Currently coodinating integration of NCO and PyNCO. Then will pass to @doutriaux1 for comments. Re: CDO, no idea if or when, though it would be similar to adding NCO.

**Comment by ThomasMaxwell** [comment]
Charlie,
   In your NCO python wrappers, are you passing the data directly to NCO in memory, or are you calling the command line operators from python?

— Tom

**Comment by czender** [comment]
Wrappers call NCO from python. Author is @jhamman and implementation is https://github.com/jhamman/nco-bindings
Comments/suggestions welcome.

**Comment by jhamman** [comment]
Exactly.  The python nco and cdo bindings are just `subprocess` wrappers that operate on netCDF files on disk.  The API's in CDO and NCO are not particularly well suited to support in memory python bindings (although that would be cool). In reality, the system call in the python bindings is typically quite cheap compared to the NCO or CDO operation so it doesn't really impact perfomance.

**Comment by doutriaux1** [comment]
@czender I thought the actual binding was what you were working on. Is it going to be "just" subprocess calls? @williams13 was it your understanding as well?

**Comment by czender** [comment]
Yes, the Python bindings in nco-bindings (aka PyNCO) are "just" subprocess calls. A library API with Python wrappers for every routine is beyond the scope of what we can do in the near term. But it's good to think big :) Anyway, this is what we have to start with. Feedback welcome.

**Comment by ThomasMaxwell** [comment]
Thanks for the feedback.   This is a great start.   I would offer that, since most of these operations are highly IO bound, having a library API supporting in-memory operations would be an excellent future goal.   In order to embed an NCO operation in an analysis pipeline one currently would have to write the input data to disk, call the NCO operator, and then read the result back from the disk.   In many cases the IO operations would take at least an order of magnitude more time then the computation.

**Comment by doutriaux1** [comment]
@ThomasMaxwell @czender the latest netcdf has the capability to write "in memory" files, that might be the best way to deal with this.

**Comment by ThomasMaxwell** [comment]
Dean,

   Is the 2015 ESGF Conference<http://www.cvent.com/events/earth-system-grid-federation-esgf-conference-2015/event-summary-89df663c495a4c03b52523abc05df8be.aspx> going to be a combined ESGF & UVCDAT F2F meeting (as in the past)?

Thanks,
Tom

**Comment by doutriaux1** [comment]
Yes @ThomasMaxwell
